code/CloudFlare_Purge.php

```diff
@@ -0,0 +1,433 @@
+     * @return mixed
+     */
+    public function handleRequest(array $data = null, $isRecursing = null, $method = 'DELETE')
+    {
+        if (array_key_exists('files', $data) && !$isRecursing) {
+            // get URL variants
+            $data['files'] = $this->getUrlVariants($data['files']);
+        }
+
+        if (array_key_exists('files', $data) && count($data['files']) > 500) {
+            // slice the array into chunks of 500 then recursively call this function.
+            // cloudflare limits cache purging to 500 files per request.
+            $chunks = ceil(count($data['files']) / 500);
+            $start = 0;
+            $responses = array();
+
+            for ($i = 0; $i < $chunks; $i++) {
+                $chunk = array_slice($data['files'], $start, 500);
+                $result = $this->handleRequest(array('files' => $chunk), true);
+                $responses[] = json_decode($result, true);
+                $start += 500;
+            }
+
+            return $responses;
+        }
+
+
+        return CloudFlare::singleton()->curlRequest($this->getEndpoint(), $data, $method);
+    }
+
+    /**
+     * Generates URL variants (Stage urls, HTTPS, Non-HTTPS)
+     *
+     * @param $urls
+     *
+     * @return array
+     */
+    public function getUrlVariants($urls)
+    {
+        $output = array();
+
+        foreach ($urls as $url) {
+            $output[] = $url;
+
+            // HTTPS Equiv
+            if (strstr($url, "http://") && !in_array(str_replace("http://", "https://", $url), $output)) {
+                $output[] = str_replace("http://", "https://", $url);
+            }
+
+            // HTTP Equiv
+            if (strstr($url, "https://") && !in_array(str_replace("https://", "http://", $url), $output)) {
+                $output[] = str_replace("http://", "https://", $url);
+            }
+        }
+
+        $this->extend("onAfterGetUrlVariants", $output);
+
+        return $output;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getEndpoint()
+    {
+        $zoneId = CloudFlare::singleton()->fetchZoneID();
+        return str_replace(":identifier", $zoneId, static::$endpoint);
+    }
+
+    /**
+     * @return bool
+     */
+    protected function isSuccessful()
+    {
+        $response = $this->getResponse();
+
+        if (!is_array($response)) {
+            return false;
+        }
+
+        if (array_key_exists("0", $response)) {
+            // multiple responses in payload, all of them need to be successful otherwise return false;
+            foreach ($response as $singular) {
+                if ($singular['success']) {
+                    continue;
+                }
+
+                return false;
+            }
+
+            return true;
+        }
+
+        if (array_key_exists('success', $response) && $response['success']) {
+            return true;
+        }
+
+        return false;
+
+    }
+
+    /**
+     * @return mixed
```

Comment:
Yeah it is array only \*glares at phpstorm phpdoc generator*. Mixed is not very helpful I personally never use it :)